src/EnergyPlus/BaseboardRadiator.cc

```diff
@@ -514,12 +514,11 @@ namespace BaseboardRadiator {
         // Do the Begin Environment initializations
         if (state.dataGlobal->BeginEnvrnFlag && this->MyEnvrnFlag && !this->SetLoopIndexFlag) {
             int WaterInletNode = this->WaterInletNode;
-            Real64 rho = state.dataPlnt->PlantLoop(this->plantLoc.loopNum).glycol->getDensity(state, Constant::HWInitConvTemp, RoutineName);
+            Real64 rho = this->plantLoc.loop->glycol->getDensity(state, Constant::HWInitConvTemp, RoutineName);
```

Comment:
This is basically all there is to this PR. In addition to `loopNum`, the PlantLocation struct now also has a pointer to the loop itself, so you can access it directly (`plantLoc.loop->X`) as opposed to indexing into the state plant object tree (`state.dataPlnt->PlantLoop(plantLoc.loopNum).X`).